draft-ietf-oauth-identity-assertion-authz-grant.md

```diff
@@ -469,6 +470,22 @@ The Resource Authorization Server's token endpoint responds with an OAuth 2.0 To
       "scope": "chat.read chat.history"
     }
 
+# Cross-Domain Client ID Handling {#client-id-mapping}
+
+There are three separate OAuth/OpenID Connect/SAML relationships involved in this flow:
+
+* Client to IdP Authorization Server (OpenID Connect or SAML)
+* Client to Resource Authorization Server (OAuth)
+* Resource Authorization Server to IdP Authorization Server (OpenID Connect or SAML)
+
+Each relationship is typically represented by independent registrations between each party. For example, the IdP Authorization Server typically issues a Client ID for the OpenID Connect Client to use for single sign-on. Similarly, the Resource Authorization Server typically issues a Client ID for the Client to use for API access to the Resource Server.
+
+In this flow, the IdP Authorization Server accepts a Token Exchange request from the Client, and issues an ID-JAG that will be consumed by the Resource Authorization Server. This means the IdP Authorization Server needs to know about the relationship between the Client and the Resource Authorization Server, in order to include a `client_id` claim in the ID-JAG that will be recognized by the Resource Authorization Server.
+
+This can be handled by the IdP Authorization Server maintaining a record of each `client_id` used between Clients and Resource Authorization Servers, which will need to be obtained by out-of-band mechanisms.
```

Comment:
```suggestion
This can be handled by the IdP Authorization Server maintaining a record of each `client_id` used between Clients and Resource Authorization Servers, which will need to be obtained by out-of-band mechanisms.  The Client still needs to authenticate using its registered credential with the Resource Authorization Server when presenting the ID-JAG for the mapped `client_id`.  Requiring a confidential client helps to prevent the IdP Authorization Server from delegating access to any client that is valid for the Resource Authorization Server.  It is still RECOMMENDED that the IdP Authorization Server authorize and audit the Client to act as another Client and not just rely on client authentication.
```

Reply:
I'm not sure I understand the "act as another client" language, but otherwise this looks good.